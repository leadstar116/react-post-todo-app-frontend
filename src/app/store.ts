import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import storage from "redux-persist/lib/storage";
import { combineReducers } from "redux";
import { persistReducer } from "redux-persist";
import { api } from "./services/api";
import auth from "../features/auth/authSlice";
import alert from "../features/alert/alertSlice";

const persistConfig = {
  key: "root",
  storage,
};

const reducers = combineReducers({
  [api.reducerPath]: api.reducer,
  auth,
  alert,
});

const persistedReducer = persistReducer(persistConfig, reducers);

export const store = configureStore({
  reducer: persistedReducer,
  devTools: process.env.NODE_ENV !== "production",
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(api.middleware),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
