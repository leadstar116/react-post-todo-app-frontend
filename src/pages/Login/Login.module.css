.loginPage {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.loginBox {
  background-color: #fff;
  display: flex;
  flex-direction: column;
  padding: 20px 20px 40px;
  width: 500px;
}

.loginBox .actionButton {
  margin: 20px 0 0 0;
}
