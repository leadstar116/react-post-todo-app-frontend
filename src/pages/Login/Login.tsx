import React, { useState, useEffect } from "react";
import { useNavigate } from "react-router-dom";
import { Box, Button, TextField } from "@mui/material";
import { useAppDispatch, useAppSelector } from "../../app/hooks";
import { selectLoggedInStatus } from "../../features/auth/authSlice";
import CircularProgressBar from "../../shared/components/CircularProgressBar/CircularProgressBar";
import styles from "./Login.module.css";
import { useLoginMutation } from "../../app/services/auth";
import { newAlert } from "../../features/alert/alertSlice";

export default function Login() {
  const navigate = useNavigate();
  const dispatch = useAppDispatch();
  const isLoggedIn = useAppSelector(selectLoggedInStatus);
  const [login, { isLoading, error }] = useLoginMutation();
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  useEffect(() => {
    if (isLoggedIn) {
      navigate("/");
    }
  }, [isLoggedIn, navigate]);

  useEffect(() => {
    if (error) {
      dispatch(
        newAlert({ message: (error as any).data.message, type: "error" })
      );
    }
  }, [error, dispatch]);

  const handleLogin = async () => {
    await login({ email, password });
  };

  const handleSignUp = () => {
    navigate("/signup");
  };

  return (
    <Box className={styles.loginPage}>
      <Box className={styles.loginBox}>
        <h1>Login</h1>
        <TextField
          label="Email"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
          type="email"
          margin="normal"
        />
        <TextField
          label="Password"
          value={password}
          onChange={(e) => setPassword(e.target.value)}
          type="password"
          margin="dense"
        />
        <Button
          variant="contained"
          color="primary"
          className={styles.actionButton}
          disabled={isLoading}
          onClick={handleLogin}
        >
          {isLoading ? <CircularProgressBar /> : "Login"}
        </Button>
        <Button
          variant="contained"
          color="secondary"
          className={styles.actionButton}
          onClick={handleSignUp}
        >
          Go to Signup
        </Button>
      </Box>
    </Box>
  );
}
