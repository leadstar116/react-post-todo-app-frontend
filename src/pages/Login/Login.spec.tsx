import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import Login from "./Login";
import { createTestStore } from "../../app/store";

describe("Login", () => {
  const store = createTestStore();

  it("should render correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
