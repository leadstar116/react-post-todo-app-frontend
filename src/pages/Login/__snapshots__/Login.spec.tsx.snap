// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
<div>
  <div
    class="loginPage MuiBox-root css-0"
  >
    <div
      class="loginBox MuiBox-root css-0"
    >
      <h1>
        Login
      </h1>
      <div
        class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Email
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            type="email"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1ftyaf0"
            >
              <span>
                Email
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-marginDense MuiTextField-root css-4mzek5-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r1:"
          id=":r1:-label"
        >
          Password
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r1:"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1ftyaf0"
            >
              <span>
                Password
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root actionButton css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Login
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root actionButton css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Go to Signup
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
